src/test/java/io/github/sskorol/data/DataSuppliers.java

```diff
@@ -32,4 +34,16 @@
                                 .readFrom(dataIn.source(), (Class<T>) dataIn.entity()))).toArray())
                 .orElseThrow(() -> new NoClassDefFoundError("No Data class found"));
     }
+
+    @SuppressWarnings("unchecked")
+    @DataSupplier(flatMap = true)
+    public <T> T getDataSet(final Method method) {
+        return (T) ofNullable(method.getDeclaredAnnotation(Data.class).entity().getDeclaredFields())
+                .map(fields -> StreamEx.of(fields)
+                        .flatMap(field -> Arrays.stream(readerOf(field.getType().getAnnotation(Source.class).source())
```

Comment:
try to decompose it... maybe would be better to split the entire pipeline into separate methods to make it more readable...